binding/form_mapping.go

```diff
@@ -136,7 +143,17 @@ func tryToSetValue(value reflect.Value, field reflect.StructField, setter setter
 }
 
 func setByForm(value reflect.Value, field reflect.StructField, form map[string][]string, tagValue string, opt setOptions) (isSetted bool, err error) {
-	vs, ok := form[tagValue]
+	var (
+		vs []string
+		ok bool
+	)
+
+	if opt.isHeader {
+		vs, ok = form[textproto.CanonicalMIMEHeaderKey(tagValue)]
```

Comment:
Good idea to implement a header mapping.

In my mind, you complicate a little the current implementation of the form mapping.

This is my suggestion of an implementation (add to your new `binding/header.go` file):
```
func mapHeader(ptr interface{}, h map[string][]string) error {
	return mappingByPtr(ptr, headerSource(h), "header")
}

type headerSource map[string][]string

var _ setter = headerSource(nil)

func (hs headerSource) TrySet(value reflect.Value, field reflect.StructField, tagValue string, opt setOptions) (isSetted bool, err error) {
	return setByForm(value, field, hs, textproto.CanonicalMIMEHeaderKey(tagValue), opt)
}
```

Reply:
Very good idea, I am going to modify it at night

Reply:
Perfect, thanks!

You can even uncomment the `//var _ setter = headerSource(nil)` line, it is [legal](https://github.com/gin-gonic/gin/blob/master/binding/form_mapping.go#L41) =)